apps/web/src/app/core/router/routeTree.ts

```diff
@@ -1,32 +1,57 @@
 import { QueryClient } from '@tanstack/react-query'
-import { createRootRouteWithContext, ParsedLocation, redirect } from '@tanstack/react-router'
+import { createRootRouteWithContext, createRoute, redirect } from '@tanstack/react-router'
 
 import { authRoutes } from 'src/app/auth/routes'
+import { AuthPagesPath } from 'src/app/auth/routes/types'
+import { useAccessTokenStore } from 'src/app/auth/store'
 import { homeRoutes } from 'src/app/home/routes'
 import { walletRoutes } from 'src/app/wallet/routes'
+import { WalletPagesPath } from 'src/app/wallet/routes/types'
 
 import { Layout } from './components/layout'
 
 export const rootRoute = createRootRouteWithContext<{ client: QueryClient }>()({
   component: Layout,
 })
 
-export const requireAuth = ({ location }: { location: ParsedLocation<object> }) => {
-  {
-    // TODO: manage user session
-    if (false as const) {
-      // Redirect them to the /welcome page, but save the current location they were
+export const publicRootRoute = createRoute({
+  getParentRoute: () => rootRoute,
+  id: 'public',
+  beforeLoad: () => {
+    const accessTokenStore = useAccessTokenStore.getState()
+    const isAuthenticated = !!accessTokenStore.accessToken
+
+    if (isAuthenticated) {
+      throw redirect({
+        to: WalletPagesPath.HOME,
+      })
```

Comment:
How does tanstack react router handles this `beforeLoad` function? If the user is coming from an public route, will they be redirected to our home page? And if they are coming from a private route, this won't get triggered, right?